src/main/python/util.py

```diff
@@ -57,6 +56,55 @@ def config_input_check(text: str, available: int, res: Enum):
             return res.INT_ERROR
 
 
+def build_docker_container(detached=True, container_path='samgomena/deepshare_worker', version='latest', memory=None, cpus=None):
+    """creates and runs a docker container from:
+    https://hub.docker.com/u/samgomena/repository/docker/samgomena/deepshare_worker
+    Returns the container ID if successful, False if not
+    """
+    docker_client = check_and_instantiate_docker_client()
+    memory += "g" # add unit
+    nano_cpus = int(cpus) * (10**9)
+    try:
+        container = docker_client.containers.run(f'{container_path}:{version}', detach=detached, nano_cpus=nano_cpus, mem_limit=memory)
+    except docker.errors.ContainerError as e:
+        print('error spinning up the container! Check docker install')
+        print(e)
+        return False
+    except docker.errors.APIError as e:
+        print('error talking to docker server API, try again later')
+        return False
+    except docker.errors.DockerException as e:
+        print('something went wrong with docker')
+        print(e)
+        return False
+    return container.short_id
+
+
+def check_and_instantiate_docker_client():
```

Comment:
this name might be too long.. we can have might be idk, like `check_docker_client` ?

Reply:
good call - renamed to just `docker_client`